gems/aws-sigv4/spec/signer_spec.rb

```diff
@@ -268,6 +268,24 @@ module Sigv4
           expect(signature.headers['authorization']).to eq('AWS4-HMAC-SHA256 Credential=akid/20120101/REGION/SERVICE/aws4_request, SignedHeaders=bar;bar2;foo;host;x-amz-content-sha256;x-amz-date, Signature=4a7d3e06d1950eb64a3daa1becaa8ba030d9099858516cb2fa4533fab4e8937d')
         end
 
+        it 'retains all custom headers from the provided request' do
```

Comment:
I want to make sure we include a test that provides the `:unsigned_headers` value as well, ensuring that we do not sign those headers. Say, `'Bar2'` or similar.

The situation I want to avoid is where we suddenly start signing banned headers, which then get modified by proxies, and then cause breaking changes down the line.